### PR TITLE
Ux tweaks

### PR DESCRIPTION
**Pull request by sleidig**
see #1527

### CODE REVIEW

File: src/app/child-dev-project/notes/dashboard-widgets/important-notes-dashboard/important-notes-dashboard.component.ts [hunk 1]
```diff
@@ -12,14 +12,14 @@ import { concat, Observable } from "rxjs";
 import { first, map } from "rxjs/operators";
 import { UntilDestroy, untilDestroyed } from "@ngneat/until-destroy";
 
-@DynamicComponent("ImportantNotesComponent")
+@DynamicComponent("ImportantNotesDashboard")
```
Comment (TheSlimvReal) [inline src/app/child-dev-project/notes/dashboard-widgets/important-notes-dashboard/important-notes-dashboard.component.ts:15]:
```suggestion
@DynamicComponent("ImportantNotesDashboard")
@DynamicComponent("ImportantNotesComponent")
```
Or we have to migrate

File: src/app/core/entity-components/entity-form/invalid-form-field.error.ts [hunk 2]
```diff
@@ -0,0 +1,9 @@
+/**
+ * Custom error indicating details of a failed form validation.
+ */
+export class InvalidFormFieldError extends Error {
+  constructor(message: string) {
+    super(message);
+    Object.setPrototypeOf(this, InvalidFormFieldError.prototype);
```
Comment (TheSlimvReal) [inline src/app/core/entity-components/entity-form/invalid-form-field.error.ts:7]: why is this necessary?

File: src/app/core/entity-components/entity-form/entity-form/entity-form.component.ts [hunk 3]
```diff
@@ -97,7 +98,9 @@ export class EntityFormComponent<T extends Entity = Entity> implements OnInit {
       this.save.emit(this.entity);
       this.switchEdit();
     } catch (err) {
-      this.alertService.addWarning(err.message);
+      if (!(err instanceof InvalidFormFieldError)) {
```
Comment (TheSlimvReal) [inline src/app/core/entity-components/entity-form/entity-form/entity-form.component.ts:101]: Should we make this consistent then? The same code is used in the `RowDetails` and the `EntitySubrecord` component.

File: src/app/core/entity-components/entity-form/entity-form.service.ts [hunk 4]
```diff
@@ -129,7 +130,9 @@ export class EntityFormService {
     form.markAllAsTouched();
     if (form.invalid) {
       const invalidFields = this.getInvalidFields(form, schema);
-      throw new Error($localize`Fields: "${invalidFields}" are invalid`);
+      throw new InvalidFormFieldError(
+        $localize`Fields: "${invalidFields}" are invalid`
```
Comment (TheSlimvReal) [inline src/app/core/entity-components/entity-form/entity-form.service.ts:134]: If we don't want to display the form fields, we might as well remove this code.

**Review by TheSlimvReal — APPROVED** [review]
Thanks for taking the time to create the storybooks. This helped a lot for the review. 
Generally looks good, just the question whether we then generalise the changes with the invalid form fields directly.

File: src/app/child-dev-project/notes/dashboard-widgets/important-notes-dashboard/important-notes-dashboard.component.ts (hunk 1 above)
Comment (sleidig) [reply src/app/child-dev-project/notes/dashboard-widgets/important-notes-dashboard/important-notes-dashboard.component.ts:15]: :+1: good point. Adding this into the existing commit as well as into the detectLegacyConfig logging

Comment (sleidig) [reply src/app/child-dev-project/notes/dashboard-widgets/important-notes-dashboard/important-notes-dashboard.component.ts:15]: hmm, actually automatic migration would be much more feasible then ... :see_no_evil:

File: src/app/core/entity-components/entity-form/invalid-form-field.error.ts (hunk 2 above)
Comment (sleidig) [reply src/app/core/entity-components/entity-form/invalid-form-field.error.ts:7]: because stackoverflow :wink: 

actually not necessary here but would be needed if we add methods to the custom class (https://stackoverflow.com/a/41429145/1473411)

File: src/app/core/entity-components/entity-form/entity-form.service.ts (hunk 4 above)
Comment (sleidig) [reply src/app/core/entity-components/entity-form/entity-form.service.ts:134]: true. Removing the translation and adding a more dev centered error implementation

File: src/app/core/entity-components/entity-form/entity-form/entity-form.component.ts (hunk 3 above)
Comment (sleidig) [reply src/app/core/entity-components/entity-form/entity-form/entity-form.component.ts:101]: thanks, good suggestion! (And we should really try to merge some of that code)

implemented this now

**Review by TheSlimvReal — APPROVED** [review]
